pkg/reconciler/cronjobsource/controller.go

```diff
@@ -50,11 +51,17 @@ func NewController(
 	cronJobSourceInformer := cronjobsourceinformer.Get(ctx)
 	eventTypeInformer := eventtypeinformer.Get(ctx)
 
+	env := &env{}
+	if err := envconfig.Process("", env); err != nil {
+		panic(fmt.Errorf("unable to process CronJobSource's required environment variables: %v", err))
```

Comment:
is there a logger.Fatal or something, instead of this panic? I think we use that typically in mains, maybe we can do the same here?

Reply:
Because this isn't a main method, I don't want to actually call fatal. Although I agree that in practice, this panic will simply bubble up to kill the process. But because this is 'library' code, I want to let the main actually decide if that is what it wants.

Reply:
Note: panic allows you to recover, fatal is insta-death